### ISSUE TITLE
Failure to handle font family with 3 words in it

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#1489]
https://github.com/google/fonts/commit/e31859c92b3cd90d86f6dfa591cbdb8d4e6babf0 should not have happened; this is a bug in some script, either 

- one of @m4rc1e 's "gattling gun" scripts, or, 

- a failure of fontbakery to detect that the METADATA.pb file was borked, because the `name` key values didn't match up

I didn't check the 2 TTF files, but if they are borked too, that's a problem :)

**Comment by felipesanches** [comment]
Font Bakery properly detects it with **com.google.fonts/test/110**:

![screenshot at 2017-08-07 21 08 59](https://user-images.githubusercontent.com/213676/29050787-ad44f676-7bb4-11e7-8767-43a8bf8445f9.png)

**Comment by felipesanches** [comment]
To be honest, there was an issue with %s being used instead of {}.
That would print a badly formated FAIL message, **but still would be detecting a FAIL**.

All instances of **%s instead of {}** were fixed at https://github.com/googlefonts/fontbakery/pull/1491

**Comment by felipesanches** [comment]
@m4rc1e may still have to review his scripts because the source of the google/fonts@e31859c issue was not yet properly identified.

As Font Bakery is behaving correctly, I'll close this issue.

**Comment by m4rc1e** [comment]
@davelab6, @felipesanches,

My scripts are ok. Here are the name tables from the fonts in [google/fonts/apache/mountainsofchristmas](https://github.com/google/fonts/tree/master/apache/mountainsofchristmas)

```
Font                                platformID    encodingID    languageID    nameID  nameString
MountainsofChristmas-Regular.ttf             1             0             0         0  Copyright (c) 2010 2011, Font Diner, Inc DBA Tart Workshop (diner@fontdiner.com) with Reserved Font Name "Mountains of Christmas"
MountainsofChristmas-Regular.ttf             1             0             0         1  Mountains of Christmas
MountainsofChristmas-Regular.ttf             1             0             0         2  Regular
MountainsofChristmas-Regular.ttf             1             0             0         3  FontDiner,IncDBATartWorkshop: Mountains of Christmas: 2011
MountainsofChristmas-Regular.ttf             1             0             0         4  Mountains of Christmas
MountainsofChristmas-Regular.ttf             1             0             0         5  Version 1.002
MountainsofChristmas-Regular.ttf             1             0             0         6  MountainsofChristmas-Regular
MountainsofChristmas-Regular.ttf             1             0             0         7  Mountains of Christmas is a trademark of Font Diner, Inc DBA Tart Workshop.
MountainsofChristmas-Regular.ttf             1             0             0         8  Font Diner, Inc DBA Tart Workshop
MountainsofChristmas-Regular.ttf             1             0             0         9  Crystal Kluge
MountainsofChristmas-Regular.ttf             1             0             0        11  http://www.fontbros.com/tartworkshop.php
MountainsofChristmas-Regular.ttf             1             0             0        12  http://www.tartworkshop.com
http://scripts.sil.org/OFLis available with a FAQ at:      0             0        13  This Font Software is licensed under the SIL Open Font License,
MountainsofChristmas-Regular.ttf             1             0             0        14  http://scripts.sil.org/OFL
MountainsofChristmas-Regular.ttf             1             0             0        18  Mountains of Christmas
MountainsofChristmas-Regular.ttf             3             1          1033         0  Copyright (c) 2010 2011, Font Diner, Inc DBA Tart Workshop (diner@fontdiner.com) with Reserved Font Name "Mountains of Christmas"
MountainsofChristmas-Regular.ttf             3             1          1033         1  Mountains of Christmas
MountainsofChristmas-Regular.ttf             3             1          1033         2  Regular
MountainsofChristmas-Regular.ttf             3             1          1033         3  FontDiner,IncDBATartWorkshop: Mountains of Christmas: 2011
MountainsofChristmas-Regular.ttf             3             1          1033         4  Mountains of Christmas
MountainsofChristmas-Regular.ttf             3             1          1033         5  Version 1.002
MountainsofChristmas-Regular.ttf             3             1          1033         6  MountainsofChristmas-Regular
MountainsofChristmas-Regular.ttf             3             1          1033         7  Mountains of Christmas is a trademark of Font Diner, Inc DBA Tart Workshop.
MountainsofChristmas-Regular.ttf             3             1          1033         8  Font Diner, Inc DBA Tart Workshop
MountainsofChristmas-Regular.ttf             3             1          1033         9  Crystal Kluge
MountainsofChristmas-Regular.ttf             3             1          1033        11  http://www.fontbros.com/tartworkshop.php
MountainsofChristmas-Regular.ttf             3             1          1033        12  http://www.tartworkshop.com
http://scripts.sil.org/OFLis available with a FAQ at:      1          1033        13  This Font Software is licensed under the SIL Open Font License,
MountainsofChristmas-Regular.ttf             3             1          1033        14  http://scripts.sil.org/OFL
MountainsofChristmas-Regular.ttf             3             1          1033        18  Mountains of Christmas

MountainsofChristmas-Bold.ttf                1             0             0         0  Copyright (c) 2010 2011, Font Diner, Inc DBA Tart Workshop (diner@fontdiner.com) with Reserved Font Name "Mountains of Christmas"
MountainsofChristmas-Bold.ttf                1             0             0         1  Mountains of Christmas
MountainsofChristmas-Bold.ttf                1             0             0         2  Bold
MountainsofChristmas-Bold.ttf                1             0             0         3  FontDiner,IncDBATartWorkshop: Mountains of Christmas Bold: 2011
MountainsofChristmas-Bold.ttf                1             0             0         4  Mountains of Christmas Bold
MountainsofChristmas-Bold.ttf                1             0             0         5  Version 1.002
MountainsofChristmas-Bold.ttf                1             0             0         6  MountainsofChristmas-Bold
MountainsofChristmas-Bold.ttf                1             0             0         7  Mountains of Christmas Bold is a trademark of Font Diner, Inc DBA Tart Workshop.
MountainsofChristmas-Bold.ttf                1             0             0         8  Font Diner, Inc DBA Tart Workshop
MountainsofChristmas-Bold.ttf                1             0             0         9  Crystal Kluge
MountainsofChristmas-Bold.ttf                1             0             0        11  http://www.fontbros.com/tartworkshop.php
MountainsofChristmas-Bold.ttf                1             0             0        12  http://www.tartworkshop.com
http://scripts.sil.org/OFLis available with a FAQ at:      0             0        13  This Font Software is licensed under the SIL Open Font License,
MountainsofChristmas-Bold.ttf                1             0             0        14  http://scripts.sil.org/OFL
MountainsofChristmas-Bold.ttf                1             0             0        18  Mountains of Christmas Bold
MountainsofChristmas-Bold.ttf                3             1          1033         0  Copyright (c) 2010 2011, Font Diner, Inc DBA Tart Workshop (diner@fontdiner.com) with Reserved Font Name "Mountains of Christmas"
MountainsofChristmas-Bold.ttf                3             1          1033         1  Mountains of Christmas
MountainsofChristmas-Bold.ttf                3             1          1033         2  Bold
MountainsofChristmas-Bold.ttf                3             1          1033         3  FontDiner,IncDBATartWorkshop: Mountains of Christmas Bold: 2011
MountainsofChristmas-Bold.ttf                3             1          1033         4  Mountains of Christmas Bold
MountainsofChristmas-Bold.ttf                3             1          1033         5  Version 1.002
MountainsofChristmas-Bold.ttf                3             1          1033         6  MountainsofChristmas-Bold
MountainsofChristmas-Bold.ttf                3             1          1033         7  Mountains of Christmas Bold is a trademark of Font Diner, Inc DBA Tart Workshop.
MountainsofChristmas-Bold.ttf                3             1          1033         8  Font Diner, Inc DBA Tart Workshop
MountainsofChristmas-Bold.ttf                3             1          1033         9  Crystal Kluge
MountainsofChristmas-Bold.ttf                3             1          1033        11  http://www.fontbros.com/tartworkshop.php
MountainsofChristmas-Bold.ttf                3             1          1033        12  http://www.tartworkshop.com
http://scripts.sil.org/OFLis available with a FAQ at:      1          1033        13  This Font Software is licensed under the SIL Open Font License,
MountainsofChristmas-Bold.ttf                3             1          1033        14  http://scripts.sil.org/OFL
MountainsofChristmas-Bold.ttf                3             1          1033        18  Mountains of Christmas Bold

```

The problem is happening in [add_font.py](https://github.com/google/fonts/blob/master/tools/add_font.py). I'm assuming it has to do with a CamelCase split function, `MountainsofChristmas --> Mountainsof Christmas`

**Comment by felipesanches** [comment]
I have just found evidence that this kind of bug also manifests itself on Font Bakery at **com.google.fonts/test/017**. It is not necessarily the source of this specific MountainsofChristmas bug at the **google/fonts** repo but it certainly is an instance of an issue that may naturally pop in the implementation of several tools due to the inherent ambiguity of splitting out the "of" term in that camel-cased name.

I'm not sure yet on how to algorithmically solve the ambiguity.

![screenshot at 2017-08-08 08 51 06](https://user-images.githubusercontent.com/213676/29070781-17182e9e-7c17-11e7-86b8-c318c6f6f78e.png)

**Comment by davelab6** [comment]
I think a solution can be wrangled out of the fact that we are using the
font filename as "the source of truth", but we do have other attestations
to the truth.

Whitelisting is also possible; this exception happened due to use of a
preposition, so we could have a simple list of prepositions and map/filter
them with endsWith() against the pieces that fall out of the camelcase
split...

**Comment by felipesanches** [comment]
One could call a family Proof Read and that would fool your proposed solution by expecting "Pro of Read".

I agree though that looking at other secondary sources of info (attestations of truth) could help in a better implementation here.

**Comment by felipesanches** [comment]
This is not trivial. It will require some more thought later.

**Comment by graphicore** [comment]
> The problem is happening in add_font.py.

The Dashboard uses the same function in the google fonts API source. So this is interesting there as well. Posting here to remind my future self :-D

**Comment by felipesanches** [comment]
This issue has now moved to these new checks:

* com.google.fonts/check/157 - "Check name table: FONT_FAMILY_NAME entries."
* com.google.fonts/check/159 - "Check name table: FULL_FONT_NAME entries."

**Comment by felipesanches** [comment]
I'm convinced now that cases like "Proof Read" are likely very rare (or not yet exist) so that I'm OK with spliting out the "of" in expected family names and then if any of these bad cases surface we can safely add them to the list of exceptions.

**Comment by felipesanches** [comment]
@davelab6 's suggestion was implemented at 17d947d
